File: Easy/1-two-sum/try-4.cpp
/*
 * Problem #1: Two Sum
 * Difficulty: Easy
 * Submission: Try 4
 * status: Accepted
 * Language: cpp
 * Date: 23/11/2024, 13:18:06
 * Link: https://leetcode.com/problems/two-sum/
 */

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int comp;
        unordered_map<int,int > d;
        for(int i=0;i<nums.size();i++){
            comp=target-nums[i];
            if(d.find(comp)!=d.end()){
                return {d[comp],i};
            }
            d[nums[i]]=i;
        }
        return {};
    }
};

File: Easy/13-roman-to-integer/try-4.py
/*
 * Problem #13: Roman to Integer
 * Difficulty: Easy
 * Submission: Try 4
 * status: Accepted
 * Language: python3
 * Date: 10/12/2024, 16:32:30
 * Link: https://leetcode.com/problems/roman-to-integer/
 */

class Solution:
    def romanToInt(self, s: str) -> int:
        d = {"I": 1, "V": 5, "X": 10, "L": 50, "C": 100, "D": 500, "M": 1000}
        i = 0
        nums = 0
        l=len(s)
        while i < l:
            if i+1 < l and d[s[i + 1]] > d[s[i]]:
                nums += d[s[i + 1]] - d[s[i]]
                i += 2

            else:
                nums += d[s[i]]
                i += 1

        return nums

